CMakeLists.txt

```diff
@@ -1,111 +1,104 @@
-cmake_minimum_required(VERSION 3.14)
+cmake_minimum_required(VERSION 3.14...3.20)
 
 project(
-    example
-    VERSION 0.1.0
-    LANGUAGES CXX
+  example
+  VERSION 0.1.0
+  LANGUAGES CXX
 )
 
+option(BUILD_SHARED_LIBS "default" YES)
+
 # ---- Warning guard ----
 
 # Protect dependents from this project's warnings if the guard isn't disabled
-set(example_warning_guard SYSTEM)
-if(example_INCLUDE_WITHOUT_SYSTEM)
-  set(example_warning_guard "")
-endif()
+## XXX why? CK
+## set(example_warning_guard SYSTEM)
+## if(example_INCLUDE_WITHOUT_SYSTEM)
+##   set(example_warning_guard "")
+## endif()
 
 # ---- Declare library ----
 
 include(GenerateExportHeader)
 
-set(headers include/example/main.h)
+include(GNUInstallDirs)
+set(EXAMPLE_INCLUDE_DIRECTORY "${CMAKE_INSTALL_INCLUDEDIR}")
```

Comment:
Setting variables all over the place without structure (which are already provided by the deliniating `----` comments) just makes an unstructured, unreadable mess.

I advise strongly against using just `CMAKE_INSTALL_INCLUDEDIR` as the include path as that can and does cause issues in some cases. More details here: `https://github.com/Dobiasd/FunctionalPlus/issues/222`